src/lib.rs

```diff
@@ -20,49 +52,93 @@ impl Cleverbot {
     /// Creates a new Cleverbot instance.
     /// * `user` - The API User.
     /// * `key` - The API Key.
-    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, String> {
+    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, CleverbotError> {
         let mut response = {
             let mut args = vec![
                 ("user", &*user),
                 ("key", &*key),
             ];
             if let Some(ref nick) = nick { args.push(("nick", &*nick)) }
-            request("https://cleverbot.io/1.0/create", &*args)
+            match request("https://cleverbot.io/1.0/create", &*args) {
```

Comment:
You could just use `try!` here since you have that `impl From<HyperError> for CleverbotError`.